### PR TITLE
feat: Filter Functionality

### PR DESCRIPTION
**Pull request by Tarandeep-s1ngh**
1. Added Filter Functionality to Product Listing Page.

2. Features added:
- Price Range Filter
- Category Filter
- Rating Filter
- Price Sort Filter

3. Minor CSS changes.

### CODE REVIEW

File: src/components/Card/Card.js [hunk 1]
```diff
@@ -16,9 +19,18 @@ export const Card = ({ items }) => {
...
-            <h3 className="semibold t1p125">{items.title}</h3>
-          </Link>
+          <div className="flex-row-wrap align-items-center justify-sb">
+            <Link to="/single-product">
+              <h3 className="semibold t1p125">{items.title}</h3>
+            </Link>
+
+            <span className="rating-badge rating-badge-ecom">
+              3.5{" "}
+              <span className="rating-fill rating-fill-ecom">
```
Comment (ragini07) [inline src/components/Card/Card.js]: Remove this hardcoded rating.

File: src/components/Card/Card.js [hunk 2]
```diff
@@ -34,9 +46,14 @@ export const Card = ({ items }) => {
...
+        className="btn-primary card-btn"
+      >
         Add to Cart
-      </Link>
+      </button>
     </div>
   );
 };
+
+// export { cartProducts };
```
Comment (Nishant-96) [inline src/components/Card/Card.js:59]: Remove unwanted code

**Review by ragini07 — APPROVED** [review]
Rest looks good

**Review by pavan6115 — APPROVED** [review]
UI looks clean 👍

File: src/components/Card/Card.js (hunk 1 above)
Comment (Tarandeep-s1ngh) [reply src/components/Card/Card.js]: Oops! Thank you for pointing it out, will fix it right away!

**Review by Nishant-96 — APPROVED** [review]
Everything Looks good. 
-I like modular approach and variable namings.
-Easy to understand
-Clean File structure

**Pull request merged**